internal/experiment/openvpn/endpoint.go

```diff
@@ -178,16 +181,6 @@ func (e endpointList) Shuffle() endpointList {
 	return e
 }
 
-// defaultOptionsByProvider is a map containing base config for
-// all the known providers. We extend this base config with credentials coming
-// from the OONI API.
-var defaultOptionsByProvider = map[string]*vpnconfig.OpenVPNOptions{
-	"riseupvpn": {
-		Auth:   "SHA512",
-		Cipher: "AES-256-GCM",
-	},
-}
-
```

Comment:
removed.
fwiw, some attachment to the exploratory code in endpointList:

I think there's some value in being able to test fixed targets even when there's no contact with backend.
it can be argued that if there's no contact then we still need credentials, and if we got credentials we can perfectly get inputs too.

Reply:
another aspect is that I think the client could/should have more weight on the shuffling and possibly resampling of targets

(i.e., it's cheaper to reorder in the client; otoh, I think the backend should give all known ports for a given IP and the client should randomize all ports probed for a single IP etc). 

but all this can be re-added in the future.

Reply:
> I think there's some value in being able to test fixed targets even when there's no contact with backend.
It can be argued that if there's no contact then we still need credentials, and if we got credentials we can perfectly get inputs too.

Yes, I think this case is a bit more complex than normal because of credentials. In general, I am okay with default inputs, but here we want to avoid default input considering that the experiment is still experimental.

> another aspect is that I think the client could/should have more weight on the shuffling and possibly resampling of targets
>
> (i.e., it's cheaper to reorder in the client; otoh, I think the backend should give all known ports for a given IP and the client should randomize all ports probed for a single IP etc).
>
> but all this can be re-added in the future if we are so inclined.

The counterpoint to this argument is that we can more easily modify the backend. I think there is still some unresolved question in the design of OONI regarding how much we want to do in the backend and how much we want to do in the client. Doing more work in the backend increases the flexibility, while doing more work in the client has the issue that clients are released less frequently. Perhaps, some solution in which we could serve scripts to clients would be the sweet spot: we can do more work in the clients and upgrade more frequently.

Reply:
I am closing this conversation. Please, reopen if needed.